### ISSUE TITLE
Enforcement of no attributeBindings on tagless components breaks usage of block-slots with ember-hook

### ISSUE THREAD

**Issue by juwara0** [ciena-blueplanet/ember-block-slots#39]
In Ember-beta, new assertions were introduced to align the behavior of handlebars to that of glimmer. The new assertions enforced that attributeBindings cannot be used on tagless components.

https://github.com/emberjs/ember.js/blame/3d25f9805cf1eb37a3f5ae6e05e3e2646b4f6175/packages/ember-htmlbars/lib/system/build-component-template.js#L47-L50

```
assert('You cannot use `attributeBindings` on a tag-less component: ' + component.toString(), (() => {
  let { attributeBindings } = component;
  return tagName !== '' || !attributeBindings || attributeBindings.length === 0;
})());
```

This causes a problem with using `ember-hook` in a testing/development environment since `ember-hook` sets attributeBindings: https://github.com/Ticketfly/ember-hook/blob/master/addon/mixins/hook.js#L17

```
export default Mixin.create({
  attributeBindings: ['_hookName:data-test'],
...
```

**Comment by juwara0** [comment]
I verified that this problem exists in Ember beta: 2.8.0-beta.3 for:
ember-block-slots version: 1.1.3
ember-hook version: 1.3.1

**Comment by juwara0** [comment]
I have a fix ready to verify but there is another issue where elementId is being set in the `super()` call of `init()` https://github.com/emberjs/ember.js/pull/14108 that needs to be merged first.

**Comment by juwara0** [comment]
This issue has been addressed buy a change in `ember-hook`: https://github.com/Ticketfly/ember-hook/pull/28/files
